library/src/main/java/com/github/gfx/android/orma/event/DataSetChangedTrigger.java

```diff
@@ -17,91 +17,32 @@
 package com.github.gfx.android.orma.event;
 
 import com.github.gfx.android.orma.Schema;
-import com.github.gfx.android.orma.Selector;
 import com.github.gfx.android.orma.annotation.Experimental;
 import com.github.gfx.android.orma.core.Database;
 
 import android.support.annotation.RestrictTo;
 
-import java.lang.ref.WeakReference;
-import java.util.HashSet;
-import java.util.Map;
-import java.util.Set;
-import java.util.concurrent.ConcurrentHashMap;
-
-import io.reactivex.Observable;
-import io.reactivex.Observer;
-import io.reactivex.subjects.PublishSubject;
-
 /**
  * Helper class for query observables. This class is NOT thread-safe.
  */
 @RestrictTo(RestrictTo.Scope.LIBRARY)
 @Experimental
-public class DataSetChangedTrigger {
-    // TODO: use well-tested concurrent weak maps
-    final Map<WeakReference<Observer<DataSetChangedEvent<?>>>, Selector<?, ?>> observerMap = new ConcurrentHashMap<>();
+public interface DataSetChangedTrigger {
 
-    Set<Schema<?>> changedDataSetInTransaction = null;
-
-    public <S extends Selector<?, ?>> Observable<DataSetChangedEvent<S>> create(S selector) {
-        PublishSubject<DataSetChangedEvent<S>> subject = PublishSubject.create();
-        register(subject, selector);
-        return subject;
-    }
+    <Model> void fire(Database db, DataSetChangedEvent.Type type, Schema<Model> schema);
 
-    @SuppressWarnings("unchecked")
-    public <S extends Selector<?, ?>> void register(Observer<DataSetChangedEvent<S>> observer, Selector<?, ?> selector) {
-        observerMap.put(new WeakReference<>((Observer<DataSetChangedEvent<?>>)(Object)observer), selector);
-    }
-
-    public <Model> void fire(Database db, DataSetChangedEvent.Type type, Schema<Model> schema) {
-        if (observerMap.isEmpty()) {
-            return;
-        }
-        if (db.inTransaction()) {
-            addChangedDataSetInTransaction(schema);
-            return;
-        }
+    void fireForTransaction();
 
-        for (Map.Entry<WeakReference<Observer<DataSetChangedEvent<?>>>, Selector<?, ?>> entry : observerMap.entrySet()) {
-            Selector<?, ?> selector = entry.getValue();
-            if (schema == selector.getSchema()) {
-                WeakReference<Observer<DataSetChangedEvent<?>>> observerRef = entry.getKey();
-                if (observerRef.get() != null) {
-                    observerRef.get().onNext(new DataSetChangedEvent<>(type, selector));
-                } else {
-                    observerMap.remove(observerRef);
-                }
-            }
-        }
-    }
+    class NoOp implements DataSetChangedTrigger {
```

Comment:
Currently, experimental data changed event feature is available only when `rxJavaSupport = true`.
I think we will implement `DataSetChangedTrigger` without RxJava before removing experimental status.